torchdata/datapipes/iter/transform/bucketbatcher.py

```diff
@@ -10,18 +13,27 @@
 T_co = TypeVar("T_co", covariant=True)
 
 
-def _in_batch_shuffle_fn(data: DataChunk):
-    d = list(data)
-    random.shuffle(d)
-    return DataChunk(d)
+@functional_datapipe("in_batch_shuffle")
+class InBatchShufflerIterDataPipe(IterDataPipe[DataChunk[T_co]]):
+    r"""
+    Shuffles each mini-batches from prior DataPipe (functional name: ``in_batch_shuffle``).
+    """
+
+    def __init__(self, datapipe: IterDataPipe[DataChunk[T_co]]):
+        self.datapipe = datapipe
+
+    def __iter__(self) -> Iterator[DataChunk[T_co]]:
+        for batch in self.datapipe:
+            random.shuffle(batch)
+            yield batch
 
 
 @functional_datapipe("bucketbatch")
 class BucketBatcherIterDataPipe(IterDataPipe[DataChunk[T_co]]):
     r"""
-    Creates mini-batches of data from sorted bucket (functional name: ``bucketbatch``). An outer
+    Creates mini-batches of data from sorted bucket. An outer
     dimension will be added as ``batch_size`` if ``drop_last`` is set to ``True``,
-    or ``length % batch_size`` for the last batch if ``drop_last`` is set to ``False``.
+    or ``length % batch_size`` for the last batch if ``drop_last`` is set to ``False``. (functional name: ``bucketbatch``).
```

Comment:
We generally need to keep "(functional name: ``bucketbatch``)" before the first `.` for the documentation to render correctly. Does the functional name show up in the table with this change?

Reply:
Let me do a quick experiment.